### PR TITLE
Issue 4112 - Added a CI test

### PR DESCRIPTION
**Pull request by aadhikar**
Bug Description: If the dbhome directory is set, eg to /dev/shm/instance
then an online backup fails because it looks for the log.000000x file
in the wring diretcory.

Relates: https://github.com/389ds/389-ds-base/issues/4112

Reviewed by: ??

### CODE REVIEW

File: dirsrvtests/tests/suites/backups/backup_test.py
Comment (Firstyear) [inline dirsrvtests/tests/suites/backups/backup_test.py]: It's not a good idea to hardcode in directory names like this because /tmp/test is something I even use and dump values into. You should look at a mktemp or other way of generating randomised folder names.

File: dirsrvtests/tests/suites/backups/backup_test.py
Comment (Firstyear) [inline dirsrvtests/tests/suites/backups/backup_test.py]: Don't hardcode timeout in start/stop because on a highly loaded CI system, 10 seconds may not be enough. There is a default value here for a reason. Please remove the timeout.

**Review by Firstyear — CHANGES_REQUESTED** [review]

**Comment by aadhikar** [comment]
@Firstyear hey I have made the changes thanks for a quick review:)

**Review by droideck — APPROVED** [review]
Looks good to me!

File: dirsrvtests/tests/suites/backups/backup_test.py
Comment (Firstyear) [inline dirsrvtests/tests/suites/backups/backup_test.py]: https://docs.python.org/3/library/tempfile.html

you don't need subprocess. Whenever possible, subprocess is a last resort, always look for a python api first.

**Review by Firstyear — CHANGES_REQUESTED** [review]

**Comment by aadhikar** [comment]
@Firstyear hey I have made the changes, I was on leave so the delay.

**Comment by Firstyear** [comment]
Looks much better @aadhikar, I'll let @droideck have a look too just in case, but otherwise Ack from me.

**Pull request merged**
